controllers/v1/programsController.js

```diff
@@ -58,59 +58,504 @@ module.exports = class Programs extends Abstract {
-          {
-            $project: {
-              externalId: 1,
-              name: 1,
-              description: 1,
-              'assessments._id': 1,
-              'assessments.externalId': 1,
-              'assessments.name': 1,
-              'assessments.description': 1,
-            },
-          },
-        ]);
+        let listOfPrograms = await programsHelper.list(
+          req.pageNo, //middleware convert req.params.page as req.PageNo
+          req.pageSize, //middleware convert req.params.linit as req.PageSize
+          req.query.searchText,
+        );
 
-        if (!programDocument) {
-          return reject({
-            status: httpStatusCode.not_found.status,
-            message: messageConstants.apiResponses.PROGRAM_NOT_FOUND,
-          });
-        }
+        listOfPrograms['result'] = listOfPrograms.data;
+
+        return resolve(listOfPrograms);
+      } catch (error) {
+        return reject({
+          status: error.status || httpStatusCode.internal_server_error.status,
+          message: error.message || httpStatusCode.internal_server_error.message,
+          errorObject: error,
+        });
+      }
+    });
+  }
+
+  /**
+  * @api {post} /samiksha/v1/programs/create Create Program
+  * @apiVersion 1.0.0
+  * @apiName create
+  * @apiGroup Programs
+  * @apiSampleRequest /samiksha/v1/programs/create
+  * @apiHeader {String} X-authenticated-user-token Authenticity token 
+  * @apiParamExample {json} Request-Body:
+  * {
+      "externalId" : "PROGID01",
+      "name" : "DCPCR School Development Index 2018-19",
+      "description" : "DCPCR School Development Index 2018-19",
+      "isDeleted" : false,
+      "resourceType" : [ 
+          "program"
+      ],
+      "language" : [ 
+          "English"
+      ],
+      "keywords" : [],
+      "concepts" : [],
+      "userId":"a082787f-8f8f-42f2-a706-35457ca6f1fd",
+      "imageCompression" : {
+          "quality" : 10
+      },
+      "components" : [ 
+          "5b98fa069f664f7e1ae7498c"
+      ],
+      "scope" : {
+          "entityType" : "state",
+          "entities" : ["5d6609ef81a57a6173a79e78"],
+          "roles" : ["HM"]
+      }
+      "startDate" : "2023-04-06T09:35:00.000Z",
+      "endDate" : ""2024-04-06T09:35:00.000Z" // optional
+    }
+  * @apiParamExample {json} Response:
+   {
+    "message": "Program created successfully",
+    "status": 200,
+    "result": {
+        "_id": "5ff09aa4a43c952a32279234"
+    }
+   } 
+  * @apiUse successBody
+  * @apiUse errorBody
+  */
+
+  /**
+   * Create program.
+   * @method {POST}
+   * @name create
+   * @param {Object} req - requested data.
+   * @returns {JSON} - created program document.
+   */
+
+  async create(req) {
+    return new Promise(async (resolve, reject) => {
+      try {
+        req.body.userId = req.userDetails.userId;
+        let programCreationData = await programsHelper.create(
+          req.body,
+          true, // checkDate
+          req.userDetails.userToken,
```

Comment:
@praveenKDass token is not required

Reply:
removing token from arg @VISHNUDAS-tunerlabs